reactor-core/src/main/java/reactor/core/publisher/Hooks.java

```diff
@@ -433,6 +434,83 @@ public static void resetOnOperatorError() {
 		}
 	}
 
+	/**
+	 * Add or replace a named scheduling {@link Function decorator}. With subsequent calls
+	 * to this method, the onSchedule hook can be a composite of several sub-hooks, each
+	 * with a different key.
+	 * <p>
+	 * The sub-hook is a {@link Function} taking the scheduled {@link Runnable}.
+	 * It returns the decorated {@link Runnable}.
+	 *
+	 * @param key the key under which to set up the onSchedule sub-hook
+	 * @param decorator the {@link Runnable} decorator to add (or replace, if key is already present)
+	 * @see #resetOnSchedule(String)
+	 * @see #resetOnSchedule()
+	 */
+	public static void onSchedule(String key, Function<Runnable, Runnable> decorator) {
```

Comment:
@bsideup I aligned the naming of the method and its behavior with other hooks. Main change is that it is now "add or replace" (`put`). Unless there is a compelling reason for this particular hook to be `putIfAbsent` unlike the others?